pkg/action/git-credential.go

```diff
@@ -0,0 +1,189 @@
+package action
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"strings"
+
+	"github.com/justwatchcom/gopass/pkg/store/sub"
+
+	"github.com/justwatchcom/gopass/pkg/store/secret"
+
+	"github.com/justwatchcom/gopass/pkg/fsutil"
+
+	"github.com/justwatchcom/gopass/pkg/out"
+
+	"github.com/justwatchcom/gopass/pkg/ctxutil"
+
+	"github.com/urfave/cli"
+)
+
+type gitCredentials struct {
+	Protocol string
+	Host     string
+	Path     string
+	Username string
+	Password string
+}
+
+func (c *gitCredentials) WriteTo(w io.Writer) (int64, error) {
+	var n int64
+	if c.Protocol != "" {
+		i, err := io.WriteString(w, "protocol="+c.Protocol+"\n")
+		n += int64(i)
+		if err != nil {
+			return n, err
+		}
+	}
+	if c.Host != "" {
+		i, err := io.WriteString(w, "host="+c.Host+"\n")
+		n += int64(i)
+		if err != nil {
+			return n, err
+		}
+	}
+	if c.Path != "" {
+		i, err := io.WriteString(w, "path="+c.Path+"\n")
+		n += int64(i)
+		if err != nil {
+			return n, err
+		}
+	}
+	if c.Username != "" {
+		i, err := io.WriteString(w, "username="+c.Username+"\n")
+		n += int64(i)
+		if err != nil {
+			return n, err
+		}
+	}
+	if c.Password != "" {
+		i, err := io.WriteString(w, "password="+c.Password+"\n")
+		n += int64(i)
+		if err != nil {
+			return n, err
+		}
+	}
+	return n, nil
+}
+
+func parseGitCredentials(r io.Reader) (*gitCredentials, error) {
+	rd := bufio.NewReader(r)
+	c := &gitCredentials{}
+	for {
+		key, err := rd.ReadString('=')
+		if err == io.EOF {
+			return c, nil
+		} else if err != nil {
+			return nil, err
+		}
+		key = strings.TrimSuffix(key, "=")
+		val, err := rd.ReadString('\n')
+		if err == io.EOF {
+			err = io.ErrUnexpectedEOF
+		}
+		if err != nil {
+			return nil, err
+		}
+		val = strings.TrimSuffix(val, "\n")
+		switch key {
+		case "protocol":
+			c.Protocol = val
+		case "host":
+			c.Host = val
+		case "path":
+			c.Path = val
+		case "username":
+			c.Username = val
+		case "password":
+			c.Password = val
+		}
+	}
+}
+
+// GitCredentialBefore is executed before another git-credential command
+func (s *Action) GitCredentialBefore(ctx context.Context, c *cli.Context) error {
+	if !ctxutil.IsStdin(ctx) {
+		return ExitError(ctx, ExitUsage, nil, "missing stdin from git")
+	}
+	return s.Initialized(ctx, c)
+}
+
+// GitCredentialGet returns a credential to git
+func (s *Action) GitCredentialGet(ctx context.Context, c *cli.Context) error {
+	cred, err := parseGitCredentials(os.Stdin)
+	if err != nil {
+		return ExitError(ctx, ExitUnsupported, err, "Error: %v while parsing git-credential", err)
+	}
+	path := "git/" + fsutil.CleanFilename(cred.Host) + "/" + fsutil.CleanFilename(cred.Username)
+	if !s.Store.Exists(ctx, path) {
```

Comment:
Some comments on this would be nice. I assume you're first checking the root store and then doing a search which should find any matching entry in a mounted substore?